src/Functions/LeastGreatestGeneric.h

```diff
@@ -114,6 +125,24 @@ class LeastGreatestOverloadResolver : public IFunctionOverloadResolver
         if (arguments.size() == 2 && isNumber(removeNullable(arguments[0].type)) && isNumber(removeNullable(arguments[1].type)))
             return std::make_unique<FunctionToFunctionBaseAdaptor>(SpecializedFunction::create(context), argument_types, return_type);
 
+        /// In sql query like `least(x, NULL)` or `greatest(x, NULL)`, which has constant NULL arguments, will make the return type is only null.
+        /// In such case, we should reconstruct the return type based on the not only null input arguments.
```

Comment:
I am surprised that nothing needs to be changed in function `getReturnTypeImpl` (l. 152ff).

Reply:
It seems this can be simply resolved by set `useDefaultImplementationForNulls` as `false` to `LeastGreatestOverloadResolver`. And the codes of l.128 ~ l.141 is not needed now.